internal/QueryProcessor.py

```diff
@@ -55,79 +21,49 @@ class QueryProcessor:
     #       -> send query to a local model (it will figure out what's important and query-worthy)
     #       -> send local model's response to google language
     def query_vision(self, query, image):
-        query_label = predict(query)
-        if query_label == "offline_question":
-            return local_llm_predict(query)
-        elif query_label == "basic_question":
-            pass
-        elif query_label == "complex_question":
-            language_response = self.query_google_language(query)
-            system_prompt = f"""
-            You will answer my question based on the following information:
-
-            1. Relevant entity information extracted from the question (language_response): {language_response}
-
-            Please synthesize the above information and accurately respond to my query. If you are uncertain about any aspects, please indicate the reasons for your uncertainty and make reasonable inferences based on the available information.
-            """
-            response = self.query_gpt(query, system_prompt)
-            return response
-        elif query_label == "vision":
-            vision_response = self.query_google_vision(image)
-            language_response = self.query_google_language(query)
-            system_prompt = f"""
-    You will answer my question based on the following information:
-    
-    1. Image label information (vision_response): {vision_response}
-    2. Relevant entity information extracted from the question (language_response): {language_response}
-    
-    Please synthesize the above information and accurately respond to my query. If you are uncertain about any aspects, please indicate the reasons for your uncertainty and make reasonable inferences based on the available information.
-    """
-            response = self.query_gpt(query, system_prompt)
-            return response
-        elif query_label == "explicit":
-            return "A query which is inappropriate "
+        vision_response = self.query_google_vision(image)
+        language_response = self.query_google_language(query)
+        combined_response = f"""
```

Comment:
This won't work (GPT won't know how to interpret this or what we want from it). It needs to either be formed as an instruction or a question.

Reply:
Can you be more specific for this "It needs to either be formed as an instruction or a question."

Reply:
Sure.

Causal LLMs (like GPT, Llama, etc) return an inferred response based on the query; this is an oversimplification, but think of their training data like questions and answers--it predicts the answer to the query based on answers to similar questions it's been trained on. When we give it something like a block of text without a clear question or instruction, it has much more difficulty predicting and returning a reasonable response.

This prompt in general also doesn't make much sense for the task--we want to find out what's in the image (send the image to google vision) then somehow merge the original user's query with google vision's response. So if the user asks "How many calories are in this?" while looking at a can of coke, we should first ask vision to figure out what the focus of the image is and to identify it, then combine that with the original query to get one like "How many calories are in a can of coke?". *That* is the point which we send a query to google lang (*not* GPT), since it'll have more accurate information than GPT (google is far more capable of providing timely and accurate information for queries like this).